### PR TITLE
Refactor usercluster-controller-manager's RBAC controller

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
`klog` should not be used in KKP. This PR removes its usage in the rbac controller and at the same time refactors it to use our reconciling framework instead of reinventing the "get-create-check-update" pattern.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/rbac/reconciler.go [hunk 1]
```diff
@@ -19,90 +19,85 @@ package rbacusercluster
...
-	err = r.ensureRBACClusterRoleBinding(ctx, resourceName)
+
+	err = r.ensureRBACClusterRoleBinding(ctx, request.Name)
 	if err != nil {
 		return err
 	}
 
+	r.rLock.Lock()
+	defer r.rLock.Unlock()
+	r.reconciledSuccessfullyOnce = true
```
Comment (kron4eg) [inline pkg/controller/user-cluster-controller-manager/rbac/reconciler.go:74]: I don't see the use of this var, what's it's for?

Comment (xrstf) [reply pkg/controller/user-cluster-controller-manager/rbac/reconciler.go:74]: Check the `registerReconciledCheck` call

**Review by kron4eg — APPROVED** [review]

**Pull request merged**